app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/deployManifestConfig.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { IScope, IController } from 'angular';
+
+import {
+  KubernetesManifestCommandBuilder
+} from '../../../manifest/manifestCommandBuilder.service';
+
+import { load } from 'js-yaml'
+
+export class KubernetesV2DeployManifestConfigCtrl implements IController {
+
+  public state = {
+    loaded: false,
+  };
+
+  constructor(private $scope: IScope,
+              private kubernetesManifestCommandBuilder: KubernetesManifestCommandBuilder) {
+    'ngInject';
+    this.kubernetesManifestCommandBuilder.buildNewManifestCommand(this.$scope.application)
+      .then((builtCommand) => {
+        this.$scope.stage.backingData = builtCommand.backingData;
+        if (this.$scope.stage.isNew) {
+          Object.assign(this.$scope.stage, builtCommand);
+        }
+        this.$scope.stage.manifest = load(this.$scope.stage.manifestText);
```

Comment:
@danielpeach we need some way to get this to be called when a pipeline is saved - how do we make that happen?

Reply:
On save sounds a little tricky, but can't you update `stage.manifest` anytime `stage.manifestText` changes?

I would add an optional `on-change` handler to your `kubernetes-manifest-entry` component, and have that trigger the updates to `stage.manifest`.